core/src/test/java/org/pentaho/platform/dataaccess/datasource/wizard/csv/CsvUtilsTest.java

```diff
@@ -324,10 +336,180 @@ public void testDateFormat7ms() {
     testAssumeColumnDetails_Dates( asList( "2015/11/10 11:57:33" ), "yyyy/MM/dd HH:mm:ss" );
   }
 
+  /**
+   * test all the custom date formats this class passes to constructor for StringEvaluator.java
+   * @throws Exception
+   */
+  @Test
+  public void testDateFormat_allFormats() throws Exception {
```

Comment:
Most of these tests in the file should probably be "integration" tests in seperate folder directory "src/it" or "src/integration" but that should be another effort.